php-classes/Emergence/Connectors/Exceptions/SyncException.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Emergence\Connectors\Exceptions;
+
+class SyncException extends \Exception
+{
+    private $message;
+    private $context;
+
+    public function __construct($message, array $context = [])
+    {
+        $this->message = $message;
+        $this->context = $context;
+
+        return parent::__construct($message);
+    }
+
+    public function getMessage($interpolate = true)
```

Comment:
@nbey boolean mode params are an antipattern, it's considered better to have differently named methods like we have already. Let's have getMessage just be a property reader for message like one would usually expect